Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java

```diff
@@ -17,79 +20,191 @@
+                updatedNodes.put(node.getConceptId(), node);
+            }
+        }
+
+        // update the links to point to the new updatedNodes
+        for (Link link: links) {
+            Node subjectNode = link.getSubject();
+            if (!(subjectNode instanceof PredicateNode)) {
+                link.setSubject(updatedNodes.get(subjectNode.getConceptId()));
+            }
+
+            Node objectNode = link.getObject();
+            if (!(objectNode instanceof PredicateNode)) {
+                link.setObject(updatedNodes.get(objectNode.getConceptId()));
+            }
+        }
+
+        // return new network with updated nodes and links
+        return new TransformedNetwork(updatedNodes, links);
+    }
+
+    @Override
+    public ITransformedNetwork getSearchTransformedNetwork(String projectId, String conceptId)
+        throws QuadrigaStorageException {
+        // get the transformed network and search for the concept id
+        List<INetwork> networkList = getNetworkList(projectId);
+
+        if (networkList == null) {
+            return null;
+        }
+
+        // get the transformed network of all the networks in a project
+        ITransformedNetwork transformedNetwork = getTransformedNetworkOfProject(projectId);
+
+        // Filter the nodes with the concept id
+        // add all the statement ids to a set
+        Set<String> statementIdSearchSet = new HashSet<String>();
+        List<Node> searchedNodes = new ArrayList<Node>();
+        for (Node node: transformedNetwork.getNodes().values()) {
+            if (conceptId.equals(node.getConceptId())) {
+                statementIdSearchSet.addAll(node.getStatementIds());
+            }
+        }
+
+        // include only those links which have statement ids in the search set
+        List<Link> finalLinks = new ArrayList<Link>();
+        for (Link link: transformedNetwork.getLinks()) {
+            if (statementIdSearchSet.contains(link.getStatementId())) {
+                // statement id match
+                // add to the final link list
+                finalLinks.add(link);
+            }
+        }
+
+        // final nodes
+        Map<String, Node> finalNodes = new HashMap<String, Node>();
+        Set<Node> addedNodes = new HashSet<Node>();
+        Integer index = 0;
+        for (Link link: finalLinks) {
+            Node subjectNode = link.getSubject();
+            Node objectNode = link.getObject();
+            // if node already added then do not add it
+            // if nodes are added twice - it would produce many
+            // nodes and less links => disconnected graph
+            if (!addedNodes.contains(subjectNode)) {
+                finalNodes.put((++index).toString(), subjectNode);
+                addedNodes.add(subjectNode);
+            }
+
+            if (!addedNodes.contains(objectNode)) {
+                finalNodes.put((++index).toString(), objectNode);
+                addedNodes.add(objectNode);
+            }
```

Comment:
I first did that way but a lot was happening in the inside for loop. I thought it was easy to read the code this way. I changed the code as per your suggestion.